estela-api/api/views/stats.py

```diff
@@ -1,35 +1,87 @@
 from collections import defaultdict
-from datetime import datetime, time
+from datetime import datetime, timedelta
 from re import findall
-from typing import List, Tuple, Union
+from typing import List, Tuple
+from json import dumps
 
 from django.db.models.query import QuerySet
 from django.utils import timezone
 from drf_yasg import openapi
 from drf_yasg.utils import swagger_auto_schema
 from rest_framework import mixins, status
 from rest_framework.decorators import action
-from rest_framework.exceptions import ValidationError
+from rest_framework.pagination import PageNumberPagination
 from rest_framework.request import Request
 from rest_framework.response import Response
 from rest_framework.serializers import ListSerializer
 
 from api import errors
 from api.exceptions import DataBaseError, InvalidDateFormatException
 from api.mixins import BaseViewSet
+from api.serializers.spider import SpiderSerializer
+from api.serializers.job import SpiderJobSerializer
 from api.serializers.stats import (
-    GetJobsStatsSerializer,
-    GlobalStatsSerializer,
-    JobsMetadataSerializer,
-    SpidersJobsStatsSerializer,
+    ProjectStatsSerializer,
+    SpidersStatsSerializer,
     StatsSerializer,
+    SpidersPaginationSerializer,
+    JobsPaginationSerializer,
 )
 from config.job_manager import spiderdata_db_client
 from core.models import Project, Spider, SpiderJob
 
 
-class StatsForDashboardMixin:
+class StatsMixin:
+    numerical_stats: dict = {
+        "items_count": 0,
+        "runtime": timedelta(seconds=0),
+        "success_rate": 0.0,
+    }
+    pages_stats: dict = {
+        "total_pages": 0,
+        "scraped_pages": 0,
+        "missed_pages": 0,
+    }
+    jobs_stats: dict = {
+        "total_jobs": 0,
+        "waiting_jobs": 0,
+        "running_jobs": 0,
+        "stopped_jobs": 0,
+        "completed_jobs": 0,
+        "in_queue_jobs": 0,
+        "error_jobs": 0,
+    }
+    status_codes_stats: dict = {
+        "status_200": 0,
+        "status_301": 0,
+        "status_302": 0,
+        "status_401": 0,
+        "status_403": 0,
+        "status_404": 0,
+        "status_429": 0,
+        "status_500": 0,
+    }
+    logs_stats: dict = {
+        "total_logs": 0,
+        "debug_logs": 0,
+        "info_logs": 0,
+        "warning_logs": 0,
+        "error_logs": 0,
+        "critical_logs": 0,
+    }
+    coverage_stats: dict = {
+        "total_items": 0,
+        "total_items_coverage": 0.0,
+    }
     stats_mapping: dict = {
+        "jobs": {
+            "waiting_jobs": SpiderJob.WAITING_STATUS,
+            "running_jobs": SpiderJob.RUNNING_STATUS,
+            "stopped_jobs": SpiderJob.STOPPED_STATUS,
+            "completed_jobs": SpiderJob.COMPLETED_STATUS,
+            "in_queue_jobs": SpiderJob.IN_QUEUE_STATUS,
+            "error_jobs": SpiderJob.ERROR_STATUS,
+        },
```

Comment:
I added this dictionaries to avoid code repetition in `summarize_jobs function`, `parse_jobs` function.